modules/ipc/include/hephaestus/ipc/zenoh/types/action_server_types.h

```diff
@@ -0,0 +1,28 @@
+//================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <cstdint>
+
+namespace heph::ipc::zenoh {
+enum class ActionServerRequestStatus : uint8_t {
```

Comment:
similar to above, this compound name can be solved using a namespace `action_server`

Reply:
Yes, I would use a folder. Looking at our internal repo, types anyway becomes cluttered quickly, so folders help. AND you solve the issue of name duplication, which naturally occurs in types.